scripts/discovery.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# PYTHON_ARGCOMPLETE_OK
+import argparse
+import argcomplete
+from tuda_workspace_scripts.print import *
+from tuda_workspace_scripts.robots import *
+from tuda_workspace_scripts.discovery import *
+
+class RobotChoicesCompleter:
+    def __call__(self, prefix, parsed_args, **kwargs):
+        complete_args = [var for var in load_robots().keys()]
+        complete_args.append("off")
+        chosen_args = getattr(parsed_args, "connections", [])
+
+        if not chosen_args:
+            complete_args.extend(["all"])
+        elif "all" in chosen_args:
+            return []
+        else:
+          complete_args = list(filter(lambda x: x not in chosen_args, complete_args))
+
+        return complete_args
+
+
+def main():
+    parser = argparse.ArgumentParser(prog="discovery", description="Allows to discover other ROS2 machines")
+    robots = load_robots()
+
+    choices = list(robots.keys())
+    choices.extend(["off","all"])    
+    first_arg = parser.add_argument(
+        "robots",
+        nargs="*",
+        choices = choices,
+        help="Select robots which should be discovered by your machine. Choose 'off' to limit discovery to the localhost or 'all' to discover all known robots."
+    )
+    first_arg.completer = RobotChoicesCompleter()
+
+    # Add optional address argument
+    parser.add_argument(
+        "--address",
+        nargs="+",
+        type=str,
+        help="Specify one or more custom addresses (e.g., IP or hostname) for discovery."
+    )
```

Comment:
If we support custom addresses wouldn't it make sense to also be able to specify the port and protocol?
E.g. IP[:PORT][/PROTO] and defaults for port and proto.
Examples:
10.10.10.1
10.10.10.2:8328/quic

Reply:
I would parse it in the specific rmw implementation and the user would have to provide the custom address in a format that is compatible with the rmw.